source/Calamari/Commands/ApplyDeltaCommand.cs

```diff
@@ -0,0 +1,131 @@
+                .PositionalArgument(deltaFilePath)
+                .PositionalArgument(tempNewFilePath)
+                .Build();
+
+            Log.Info("Applying delta to {0} with hash {1} and storing as {2}", basisFilePath, fileHash,
+                newFilePath);
+
+            var result = commandLineRunner.Execute(octoDiff);
+            if (result.ExitCode != 0)
+            {
+                fileSystem.DeleteFile(tempNewFilePath, DeletionOptions.TryThreeTimes);
+                throw new CommandLineException(executable, result.ExitCode, result.Errors);
+            }
+
+            File.Move(tempNewFilePath, newFilePath);
+
+            if (!File.Exists(newFilePath))
+                throw new CommandException("Failed to apply delta file " + deltaFilePath + " to " +
+                                           basisFilePath);
+
+            var package = packageStore.GetPackage(newFilePath);
+            if (package == null) return 0;
+
+            using (var file = new FileStream(package.FullPath, FileMode.Open, FileAccess.Read, FileShare.Read))
+            {
+                var size = file.Length;
+                Log.ServiceMessages.DeltaVerification(package.FullPath, package.Metadata.Hash, size);
+            }
+
+            return 0;
+        }
+
+        void ValidateParameters(out string basisFilePath, out string deltaFilePath, out string newFilePath)
+        {
+            Guard.NotNullOrWhiteSpace(basisFileName, "No basis file was specified. Please pass --basisFileName MyPackage.1.0.0.0.nupkg");
+            Guard.NotNullOrWhiteSpace(fileHash, "No file hash was specified. Please pass --fileHash MyFileHash");
+            Guard.NotNullOrWhiteSpace(deltaFileName, "No delta file was specified. Please pass --deltaFileName MyPackage.1.0.0.0_to_1.0.0.1.octodelta");
+            Guard.NotNullOrWhiteSpace(newFileName, "No new file name was specified. Please pass --newFileName MyPackage.1.0.0.1.nupkg");
+
+            basisFilePath = Path.GetFullPath(basisFileName);
+            deltaFilePath = Path.GetFullPath(deltaFileName);
+            newFilePath = Path.Combine(packageStore.GetPackagesDirectory(), newFileName + "-" + Guid.NewGuid());
+            if (!File.Exists(basisFilePath)) throw new CommandException("Could not find basis file: " + basisFileName);
+            if (!File.Exists(deltaFilePath)) throw new CommandException("Could not find delta file: " + deltaFileName);
+
+            var previousPackage = packageStore.GetPackage(basisFilePath);
+            if (previousPackage.Metadata.Hash != fileHash)
+            {
+                throw new CommandException("Basis file hash " + previousPackage.Metadata.Hash +
+                                           " does not match the file hash specified " + fileHash);
+            }
+        }
+
+        string FindOctoDiffExecutable()
+        {
+            var basePath = Path.GetDirectoryName(GetType().Assembly.FullLocalPath());
```

Comment:
I had an issue with this method the other day, as it uses the `Assembly.CodeBase` property.  This gave me issues when the path was a UNC path. More likely than not, we actually want `Assembly.Location`
See the remarks section of https://msdn.microsoft.com/en-us/library/system.reflection.assembly.codebase%28v=vs.110%29.aspx